docs/GettingStarted.md

```diff
@@ -269,4 +276,81 @@ Patches are submitted to LLVM/MLIR via GitHub pull-requests, the basic flow is a
 6) Publish the branch on your fork of the repository and create a GitHub pull-request.
 
 When your review converges and your patch is approved, it can be merged directly on GitHub.
-If you have commit access, you can do this yourself, otherwise a reviewer can do it for you.
+If you have commit access, you can do this yourself, otherwise a reviewer can do it for you.  
+  
+## Writing a basic CIRCT Pass  
+Passes can be added at several levels in CIRCT. Here we illustrate this with a simple pass, added 
+to the `firrtl` dialect, that replaces all of the wire names with `foo`: 
+1) Update your dialect’s Passes.td file to define what your pass will do in a high-level, well documented way, 
+e.g. in `include/circt/Dialect/FIRRTL/Passes.td`:  
+```
+def FooWires : Pass<"firrtl-foo-wires", "firrtl::CircuitOp"> {
+  let summary = "Change all wires' name to foo_<n>.";
+  let description = [{
+    Very basic pass that numbers all of the wires in a given module.
+    The wires' names are then all converte to foo_<that number>.
+  }];
+  let constructor = "circt::firrtl::createFooWiresPass()";
+  let statistics = [
+    Statistic<"numChangedWires", "num-changed-wires", "Number of wires changed">
+  ];
+}
+```    
+Once this is added, compile CIRCT. This will generate a base class for your pass following the naming
+defined in the tablegen description. This base class will contain some methods that need to be implemented.
+Your goal is to now implement those.  
```

Comment:
maybe add a link to https://mlir.llvm.org/docs/PassManagement/#tablegen-specificationd